### PR TITLE
msft-golang: upgrade version 1.22.3 -> 1.22.4 to address CVE-2024-24790

### PR DESCRIPTION
**Pull request by mfrw**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] If you are adding/removing a .spec file that has multiple-versions supported, please add [@microsoft/cbl-mariner-multi-package-reviewers](https://github.com/orgs/microsoft/teams/cbl-mariner-multi-package-reviewers) team as reviewer [(Eg. golang has 2 versions 1.18, 1.21+)](https://github.com/microsoft/azurelinux/tree/2.0/SPECS/golang)
- [ ] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Upgrade version 1.22.3 -> 1.22.4 to address CVE-2024-24790

###### Change Log  

- Upgrade version 1.22.3 -> 1.22.4

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

Related PR

- #9097

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2024-24790

###### Test Methodology

- Pipeline build id: [PR-9352](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=582660&view=results)

### CODE REVIEW

File: SPECS/msft-golang/msft-golang.spec [hunk 1]
```diff
@@ -153,6 +153,9 @@ fi
 %{_bindir}/*
 
 %changelog
+* Wed Jun 07 2024 Muhammad Falak <mwani@microsoft.com> - 1.22.4-1
```
Comment (mfrw) [inline SPECS/msft-golang/msft-golang.spec:156]:
```suggestion
* Fri Jun 07 2024 Muhammad Falak <mwani@microsoft.com> - 1.22.4-1
```

**Comment by Redent0r** [comment]
For my learning, do we not need to bump the release of every package that builds with go? Like we do in https://github.com/microsoft/azurelinux/pull/9097/files#diff-56c720298bdd5df9c3dbbf9d7b75cfaf785b56a39167d43a97d8ed9a8d7dc22c

**Comment by mfrw** [comment]
> For my learning, do we not need to bump the release of every package that builds with go? Like we do in https://github.com/microsoft/azurelinux/pull/9097/files#diff-56c720298bdd5df9c3dbbf9d7b75cfaf785b56a39167d43a97d8ed9a8d7dc22c

Yes you are correct; but none of the packages depend on msft-golang.

**Comment by Redent0r** [comment]
> > For my learning, do we not need to bump the release of every package that builds with go? Like we do in https://github.com/microsoft/azurelinux/pull/9097/files#diff-56c720298bdd5df9c3dbbf9d7b75cfaf785b56a39167d43a97d8ed9a8d7dc22c
> 
> Yes you are correct; but none of the packages depend on msft-golang.

Thanks! I did also notice the buddy build [failed](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=582660&view=logs&j=8fc237db-d9de-5497-d5bc-7b55eebfb9b4&t=95de2199-5f24-5bb6-ce75-dd26f99e8922&l=64) the install tests. Are these failures safe to ignore or need to update something?

**Review by christopherco — APPROVED** [review]

**Comment by mfrw** [comment]
> > > For my learning, do we not need to bump the release of every package that builds with go? Like we do in https://github.com/microsoft/azurelinux/pull/9097/files#diff-56c720298bdd5df9c3dbbf9d7b75cfaf785b56a39167d43a97d8ed9a8d7dc22c
> > 
> > 
> > Yes you are correct; but none of the packages depend on msft-golang.
> 
> Thanks! I did also notice the buddy build [failed](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=582660&view=logs&j=8fc237db-d9de-5497-d5bc-7b55eebfb9b4&t=95de2199-5f24-5bb6-ce75-dd26f99e8922&l=64) the install tests. Are these failures safe to ignore or need to update something?

These are safe to ignore :)
Thanks for being vigilant.

**Pull request merged**